<launch>
    <arg name="use_guidance" default="false"/>

    <group ns="rocket">
        <rosparam file="$(find drone_gnc)/config/rocket_parameters.yaml"/>
    </group>

    <group ns="control/mpc">
        <rosparam file="$(find drone_gnc)/config/control_tracking_mpc_parameters.yaml"/>
    </group>

    <group ns="guidance/mpc">
        <rosparam file="$(find drone_gnc)/config/guidance_mpc_parameters.yaml"/>
    </group>

    <group ns="navigation">
        <rosparam file="$(find drone_gnc)/config/filter_parameters.yaml"/>
    </group>

<!--       <node name="guidance" pkg="drone_gnc" type="fixed_guidance" machine="gnc_machine" cwd="node" output="screen">-->
<!--           <rosparam param="speed">0.35</rosparam>-->
<!--           <rosparam param="scaling">1</rosparam>-->
<!--           <rosparam param="rotation_angle">0</rosparam>-->
<!--       </node>-->

<!--    valgrind &#45;&#45;track-origins=yes &#45;&#45;leak-check=full-->

    <group if="$(arg use_guidance)">
        <node name="guidance" pkg="drone_gnc" type="drone_guidance" machine="gnc_machine" output="screen"
              launch-prefix="taskset -c 2">
            <rosparam param="target_apogee">[0, 0, 4]</rosparam>
            <rosparam param="target_land">[0, 1, 0]</rosparam>
        </node>
    </group>

    <node name="control" pkg="drone_gnc" type="drone_control" machine="gnc_machine" output="screen"
          launch-prefix="taskset -c 3">
        <rosparam param="target_apogee">[0,0,1]</rosparam>
        <rosparam param="track_guidance" subst_value="true">$(arg use_guidance)</rosparam>
    </node>

    <node name="drone_fsm" pkg="drone_gnc" type="drone_fsm" machine="gnc_machine" output="screen">
        <rosparam param="initial_state">"Idle"</rosparam>
    </node>

    <node name="navigation" pkg="drone_gnc" type="drone_navigation" machine="gnc_machine"
          output="screen" launch-prefix="taskset -c 1">
        <rosparam param="initial_state">[0,0,0, 0,0,0, 0,0,0,1, 0,0,0]</rosparam>
        <rosparam param="estimate_params">true</rosparam>
        <rosparam param="init_estimated_params">true</rosparam>
        <rosparam param="use_gps">true</rosparam>
    </node>
</launch>


